### ISSUE TITLE
wildcard inputs broken

### ISSUE THREAD

**Issue by WASasquatch** [comfyanonymous/ComfyUI#257]
Wildcard inputs allow you to input to anything as they should, but ComfyUI is trying to validate them, where `*` won't equal the nodes type.

**Comment by comfyanonymous** [comment]
Can you give me an example where it would make sense to have wildcard inputs in the backend?

**Comment by WASasquatch** [comment]
> Can you give me an example where it would make sense to have wildcard inputs in the backend?

Conversion of types, debug anything and pass it along as already noted, nodes that handle multiple types of input such as saving stuff to disk, etc.

**Comment by WASasquatch** [comment]
One of the most important facets is I am incorporating logic, and conditions, and that means they should be able to handle, and output any type.

**Comment by comfyanonymous** [comment]
Ok I see then that makes sense.

**Comment by WASasquatch** [comment]
> Ok I see then that makes sense.

Thinking further, and I'm not positive if it's positive, but a method to dynamically change the output type seems like a better option. My reasoning is this: it simply eliminates a lot of the risk of unintentional user error. Cause then, say, my logic node could have a mode `output_type` and they can select their desired input/out, or input + output conversion. This would prevent hooking up the wrong nodes and triggering possibly catastrophic errors.

**Comment by kenjiqq** [comment]
This would be very useful, i was making a list iterator, and having to make separate node for every type of list isn't very flexible. NumberListIterator,TextListIterator, ImageListIterator, LatentListIterator etc etc

**Comment by WASasquatch** [comment]
Yeah we should be able to assume we know what we're doing and handle wildcards types with is instance checking or `type(input) == list` etc
